oat-sa/tao-core
# Feature/tao 10453/implement query builder

**related to:** https://oat-sa.atlassian.net/browse/TAO-10454

This PR replaces the current `search` component by the new `searchModal` component, and is dependant of [this](https://github.com/oat-sa/tao-core-ui-fe/pull/172) PR on `tao-core-ui-fe`.

**How to test**
To be able to test this PR locally you need to setup your TAO instance to manage the new Advanced search feature. To do so you need to:

- On your root `composer.json` set the following dependencies:
```
    "oat-sa/tao-core": "feature/TAO-10453/implement-query-builder",
    "oat-sa/generis": "dev-feature/TAO-10203-advanced-search as 12.33.2",
    "oat-sa/extension-tao-dac-simple": "dev-feature/TAO-10203-advanced-search",
    "oat-sa/lib-tao-elasticsearch": "dev-feature/TAO-10203-advanced-search as 1.1.1"
```
- run `composer install`
- if you work with `taoDockerize` run `taoDockerize install`  (database will be reset)
- If you do not user `taoDockerize` run `php tao/scripts/taoUpdate.php`

Notice that because of its dependency, on `views/package.json` `@oat-sa/tao-core-ui` needs to be pointing to `feature/TAO-10454/implement-search-retention` following your usual way to do it (linking package to local repository, pointing `package.json` to origin branch...).

Now `searchModal` component should be used on search trigger instead of previous search component. Notice that advanced search syntax is not enabled if you do not set up an elastic search server, which is not required to test this feature.

Please let me know if you need more details about how to test the feature.

![search-retention](https://user-images.githubusercontent.com/14041944/91667796-6af59800-eb07-11ea-8fcd-ce2c3feccbe3.gif)
